service/perfdiscovery.go

```diff
@@ -5,6 +5,13 @@ import (
 )
 
 func (uis *UIServer) perfdiscoveryPage(w http.ResponseWriter, r *http.Request) {
+	projCtx := MustHaveProjectContext(r)
+	project, err := projCtx.GetProject()
```

Comment:
I think this check is unnecessary - the middleware already checks that the project exists

Reply:
Ha apparently the middleware doesn't error if the project doesn't exist. This is fine then